src/webgpu/shader/execution/expression/expression.ts

```diff
@@ -563,159 +563,349 @@ function packScalarsToVector(
 }
 
 /**
- * Generates a Case for the param and unary interval generator provided.
+ * Builds a Case for the param and unary interval generator provided.
  * The Case will use use an interval comparator for matching results.
  * @param param the param to pass into the unary operation
+ * @param check_intervals should intervals be checked if they are f32 finite
  * @param ops callbacks that implement generating an acceptance interval for a unary operation
  */
-export function makeUnaryToF32IntervalCase(param: number, ...ops: PointToInterval[]): Case {
+function makeUnaryToF32IntervalCase(
+  param: number,
+  check_intervals: boolean,
```

Comment:
instead of a `boolean` here, how about we use TypeScript's string enumerators so that the call sites are more readable (e.g. `'full' | 'finite_only'`)?  It would also be more future-proof, if we wanted to add more filter modes.
Alternatively, you could change this to be a predicate function, removing the embedded logic altogether.

Reply:
Done